crossfire/clients/__init__.py

```diff
@@ -127,5 +113,19 @@ def occurrences(
             max_parallel_requests=max_parallel_requests,
             format=format,
         )
+        return await occurrences()
+
+
+class Client(AsyncClient):
+    def __init__(self, email=None, password=None, max_parallel_requests=None):
+        super().__init__(
+            email=email, password=password, max_parallel_requests=max_parallel_requests
+        )
+
+    async def _states(self, format=None):
+        return await super().states(format=format)
+
+    def states(self, format=None):
         loop = get_event_loop()
-        return loop.run_until_complete(occurrences())
+        states, _ = loop.run_until_complete(self._states(format=format))
```

Comment:
Então agora [a suíte de testes aponta um erro aqui](https://github.com/FelipeSBarros/crossfire/actions/runs/7020309405/job/19099697720#step:6:57).

Segundo o erro, a gente espera dois elementos em sequência como retorno de `loop.run_until_complete(self._states(format=format))`, mas esse retorno não tem elemento algum. Estranho.

O que eu faria?

1. Colocaria um debugger `import pytest; pytest.set_trace()` antes dessa linha
2. Rodaria apenas o teste em questão com `pytest -k test_client_load_states`
3. No debugger tentaria ver qual o resultado de `self._states()` e `super().states()` para entender se estamos _mockando_ corretamante

Faz sentido?

Reply:
Faz. Comecei a implementar sua sugestão. Mas pela falta de experiencia com `pdb`, terei que recorrer ao @dunosauro e a [live de python sobre dubugger](https://www.youtube.com/watch?v=yffiyHEiUvo).
Espero voltar em breve com alguma evolução. Mas talvez demore.

Reply:
Conhecer o `pdb` é legal, mas o que você precisa agora é o básico. Você pode imaginar que ele é um console como o `python` ou `ipython`.

Digitar `self._states` vai te mostrar o que é esse objeto. Digitar `super().states` idem.

Você pode chamar as funções (mas como elas são assíncronas não vão retornar) adicionando os parênteses.

Só isso já é o suficiente para o que eu sugeri de averiguar se estamos _mockando_ o que pensamos que estamos : )

Reply:
> Conhecer o `pdb` é legal, mas o que você precisa agora é o básico. Você pode imaginar que ele é um console como o `python` ou `ipython`.
> 
> Digitar `self._states` vai te mostrar o que é esse objeto. Digitar `super().states` idem.
> 
> Você pode chamar as funções (mas como elas são assíncronas não vão retornar) adicionando os parênteses.
> 
> Só isso já é o suficiente para o que eu sugeri de averiguar se estamos _mockando_ o que pensamos que estamos : )

OK. Entendo que é para rodar isso e retornar aqui. Não para commitar essas mudanças, já que é um processo de debug.
Em breve retorno com essas infos.

Reply:
Exato. Você pode, também, executar a linha que dá o erro, sem o `states, _ =` (pois é isso que faz o _unpack_ que gera o erro) com `loop.run_until_complete(self._states(format=format))` e ver se ajuda : )

Reply:
```python
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /mnt/Trabalho/repos/crossfire/crossfire/clients/__init__.py(132)states()
-> states, _ = loop.run_until_complete(self._states(format=format))
(Pdb) self._states()
<coroutine object Client._states at 0x7fa361753840>
(Pdb) super().states()
*** RuntimeError: super(): no arguments

```

Dei uma procurada e vi [genta falando](https://stackoverflow.com/a/66017460) que:
> I got this same error when trying to call super while in a pdb shell and ended up going down a rabbit hole trying to figure it out. You need to add the parent class you want to call super on and self to the call in order for it to run - super(<ParentClass>, self) - in pdb. Or at least just know that super won't work as expected in pdb.

Tentei da forma indicada:
```python
(Pdb) super(AsyncClient, self).states()
*** AttributeError: 'super' object has no attribute 'states'
```

Reply:
> Exato. Você pode, também, executar a linha que dá o erro, sem o `states, _ =` (pois é isso que faz o _unpack_ que gera o erro) com `loop.run_until_complete(self._states(format=format))` e ver se ajuda : )

```python
(Pdb) loop.run_until_complete(self._states(format=format))
<AsyncMock name='states()' id='140339737342352'>
```

Reply:
Interessante: Ao rodar o `self._states()` ele retorna `<coroutine object Client._states at 0x7fa361753840>`
Ao rodar `loop.run_until_complete(self._states(format=format))` ele retorna que estamos rodando um mock de AsyncClient...:
`<AsyncMock name='states()' id='140339737342352'>`.
Era esperado isso?

Reply:
Como te disse no Telegram:

> Pode ser herdando sim. Vai ser mais mais complicado pois vais ter q usar o `super()` — mas fica um código lindão.
>
> A alternativa (mais fácil de escrever) seria simplesmente um `__init__` que não espera argumento algum, e que instancia `AsyncClient` como um `self.client`

Por mim, dá para ignorar essa complexidade do `super()` por agora e explorar o [resultado da operação que gera o erro](https://github.com/FelipeSBarros/crossfire/pull/74#discussion_r1408132645).

Reply:
> Era esperado isso?

Sim, dado que [você _mockou_ o método `states` do `AsyncClient`](https://github.com/FelipeSBarros/crossfire/blob/a54bade7c0f257ef13a2e9cdf1988b38b8923af6/tests/test_client.py#L249), dado que [o `self._states` chama justamente o `states` do `AsyncClient`](https://github.com/FelipeSBarros/crossfire/blob/a54bade7c0f257ef13a2e9cdf1988b38b8923af6/crossfire/clients/__init__.py#L126), acho esperado…

Reply:
> Por mim, dá para ignorar essa complexidade do `super()` por agora

Por "ignorar" não seria desfazer a classe  e métodos como estão (usand super() ) para usar a sua alternativa. Confere?

A operação que gera o erro é a que foi mockada. `AsyncClient.states`. Que é executada em `loop.run_until_complete(self._states(format=format))`.
Logo, estou chegando a conclusão de que o erro (
`E       ValueError: not enough values to unpack (expected 2, got 0)` se dá pelo fato de eu não ter definido um `return_value`. Faz sentido isso?

Reply:
> Faz sentido isso?

Total : )

Reply:
Pronto. Adicionei. Como estamos apenas testando que o `AsyncClient.states` está sendo chamado, coloquei um `Return_value` qualquer, só para que o método mencionado possa ser executado.
Está bem assim, ou deveria simular a resposta? Acho que isso já temos testado, não?
Me avisa assim vejo de seguir aos métodos que faltam...

Reply:
Uhuu… [CI verdinha agora](https://github.com/FelipeSBarros/crossfire/actions/runs/7046694947) : )

> Está bem assim, ou deveria simular a resposta? Acho que isso já temos testado, não?

Exato, isso já temos testado.